awx/main/scheduler/task_manager.py

```diff
@@ -600,13 +606,9 @@ def process_pending_tasks(self, pending_tasks):
             found_acceptable_queue = False
 
             preferred_instance_groups = self.instance_groups.get_instance_groups_from_task_cache(task)
-            # Determine if there is control capacity for the task
-            if task.capacity_type == 'control':
-                control_impact = task.task_impact + settings.AWX_CONTROL_NODE_TASK_IMPACT
-            else:
-                control_impact = settings.AWX_CONTROL_NODE_TASK_IMPACT
+
             control_instance = self.instance_groups.fit_task_to_most_remaining_capacity_instance(
-                task, instance_group_name=settings.DEFAULT_CONTROL_PLANE_QUEUE_NAME, impact=control_impact, capacity_type='control'
+                task, instance_group_name=settings.DEFAULT_CONTROL_PLANE_QUEUE_NAME, impact=settings.AWX_CONTROL_NODE_TASK_IMPACT, capacity_type='control'
```

Comment:
I could be wrong, but I think you should back out changes to these lines, but leave everything else. Without this, control actions (like project updates) can put a node over capacity.

Reply:
no I think you're right -- for capacity type "control", we should be fitting with the task_impact + control_node_task_impact